app/services/analytics.rb

```diff
@@ -6,18 +6,21 @@ def initialize(user, request)
 
   def track_event(event, attributes = {})
     analytics_hash = {
-      event: event,
-      properties: attributes.except(:user_id),
+      event_properties: attributes.except(:user_id),
       user_id: attributes[:user_id] || uuid
     }
 
-    ANALYTICS_LOGGER.info(analytics_hash.merge!(request_attributes))
+    ahoy.track(event, analytics_hash.merge!(request_attributes))
   end
 
   private
 
   attr_reader :user, :request
 
+  def ahoy
+    @ahoy ||= Rails.env.test? ? FakeAhoyTracker.new : Ahoy::Tracker.new(request: request)
+  end
```

Comment:
@zachmargolis I know you had made a comment about this technique in a previous PR, where you suggested to use `stub_const` in tests, but it caused the tests to fail and caused me some head scratching due to an error message that didn't make sense and included the Ahoy gem in the stack trace.

The problem was that I was stubbing `FakeAhoyTracker` for `Ahoy::Tracker`, and because I didn't have an `initialize` method defined in `FakeAhoyTracker`, it complained about wrong number of arguments, pointing to Ahoy's [controller initializer](https://github.com/ankane/ahoy/blob/master/lib/ahoy/controller.rb#L20). It said 1 argument was passed, but 0 were expected. That didn't make sense because `Ahoy::Tracker` clearly accepts arguments. Finally it dawned on me that the tests were using `FakeAhoyTracker`, and stubbing the const meant that I had to implement all the methods called by the real class in my fake class.

The whole point of the fake class is so that the tests use that class and only that class, and it shouldn't be necessary to implement all the methods from the real class in the fake class. I don't want the tests to touch the real class at all, which is why I prefer this way of doing it.